SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/LayeredLBStrategy.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "Utils/LayerInfo.h"
+#include "Utils/SpatialActorUtils.h"
+
+#include "Templates/Tuple.h"
+
+DEFINE_LOG_CATEGORY(LogLayeredLBStrategy);
+
+ULayeredLBStrategy::ULayeredLBStrategy()
+	: Super()
+{
+}
+
+ULayeredLBStrategy::~ULayeredLBStrategy()
+{
+	for (const auto& Elem : LayerNameToLBStrategy)
+	{
+		Elem.Value->RemoveFromRoot();
+	}
+}
+
+void ULayeredLBStrategy::Init()
+{
+	Super::Init();
+
+	VirtualWorkerId CurrentVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID + 1;
+
+	if (const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>())
```

Comment:
Would prefer to grab settings directly rather than having indentation. Could add a check() but as you say things would've blown up much earlier if Settings wasn't valid

Reply:
Yup, reducing indentation good.